base/namedtuple.jl

```diff
@@ -0,0 +1,91 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+@generated function namedtuple(::Type{NamedTuple{names,T} where T}, args...) where names
```

Comment:
This could perhaps just be simplified to
```julia
@generated function Base.namedtuple(syms::NTuple{N, Symbol}, args...) where N
    if length(args) == N
        Expr(:new, :(NamedTuple{syms,$(Tuple{args...})}), Any[ :(args[$i]) for i in 1:N ]...)
    else
        :(throw(ArgumentError("wrong number of arguments to named tuple constructor")))
    end
end
```
since it's a bit redundant to pass `NamedTuple` to the `namedtuple` function. Also can this be exported? I'm finding it the handiest way to generate non-literal NamedTuples. It has a nice companion in `tuple(args...)`, you just have to pass in the fieldnames as well.